guide/training_with_augmentations.ipynb

```diff
@@ -0,0 +1,360 @@
+    "    features_labels=train_paths,\n",
+    "    filename_template='data/data-train_shard-{shard:03d}.tfrec',\n",
+    "    examples_per_shard=3)\n",
+    "\n",
+    "nobrainer.tfrecord.write(\n",
+    "    features_labels=evaluate_paths,\n",
+    "    filename_template='data/data-evaluate_shard-{shard:03d}.tfrec',\n",
+    "    examples_per_shard=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!ls data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create Datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_classes = 1\n",
+    "batch_size = 2\n",
+    "volume_shape = (256, 256, 256)\n",
+    "block_shape = (128, 128, 128)\n",
+    "n_epochs = None\n",
+    "shuffle_buffer_size = 10\n",
+    "num_parallel_calls = 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Take a look at different augmentation options in Nobrainer spatial and intensity transforms. To set training with multiple augmentations, the 'augment' will be set as a list where their order will determine the sequence of execution. For example augment option below will first add Gaussian noise and will then perform the random flip. \n",
+    "Parameters of any given augmentation techniques can be set as shown below ( eg. 'noise_mean':0.1') otherwise default parameter settings will be applied. \n",
+    "\n",
+    "For training without augmentation, set 'augment = None'. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nobrainer.intensity_transforms import *\n",
+    "from nobrainer.spatial_transforms import *\n",
+    "\n",
+    "augment = [(addGaussianNoise, {'noise_mean':0.1,'noise_std':0.5}), (randomflip_leftright)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset_train = nobrainer.dataset.get_dataset(\n",
+    "    file_pattern='data/data-train_shard-*.tfrec',\n",
+    "    n_classes=n_classes,\n",
+    "    batch_size=batch_size,\n",
+    "    volume_shape=volume_shape,\n",
+    "    block_shape=block_shape,\n",
+    "    n_epochs=n_epochs,\n",
+    "    augment=augment,\n",
+    "    shuffle_buffer_size=shuffle_buffer_size,\n",
+    "    num_parallel_calls=num_parallel_calls,\n",
+    ")\n",
+    "\n",
+    "dataset_evaluate = nobrainer.dataset.get_dataset(\n",
+    "    file_pattern='data/data-evaluate_shard-*.tfrec',\n",
+    "    n_classes=n_classes,\n",
+    "    batch_size=batch_size,\n",
+    "    volume_shape=volume_shape,\n",
+    "    block_shape=block_shape,\n",
+    "    n_epochs=1,\n",
+    "    augment=False,\n",
+    "    shuffle_buffer_size=None,\n",
+    "    num_parallel_calls=1,\n",
+    ")"
```

Comment:
It is useful to have some data visualization after augmentation is applied.

Reply:
right but I tried to do this as an on-the-fly augmentation (while training). I can add another tutorial for offline augmentation ?

Reply:
got your point. yes we can do that separately just to show what each augmentation does.